### ISSUE TITLE
IOS Problem with data only message

### ISSUE THREAD

**Issue by ikiugu** [evollu/react-native-fcm#791]
Thanks @evollu for this library. I have been struggling with getting push to work with FCM and ios for a while. I followed your instructions and when I was done, I could send messages from Firebase Console and I would receive them in the app. Here is my problem though, when I send the following using the FCM api, it works
`{ 
 "to": "token-here",
 "priority": "high",
 "notification": {
    "title": "You have a new message",
    "text": "The new message here"
  },
 "data": {
     "customId": "<my custom id>",
     "badge": 0,
    "alert": "New data is available"
  }
}`
but when I remove the notifications property, like this
`{ 
 "to": "token-here",
 "priority": "high",
 "data": {
     "customId": "<my custom id>",
     "badge": 0,
     "sound": "cheering.caf",
    "alert": "New data is available"
  }
}`
the app does not display anything. Even the breakpoints in `FCM.on(FCMEvent.Notification, async (notif) =>` are not being hit. What could I be doing wrong?

**Comment by Ikiugu** [comment]
This is working well but only in foreground. I am using the sample project and the messages are 'stored' when the app is in background/killed then when I open it, they display. How would I show them even when app is killed/background?

**Comment by evollu** [comment]
iOS? add `content_available: true`

**Comment by Ikiugu** [comment]
I have added that and it still isn't visible

**Comment by evollu** [comment]
`I am using the sample project and the messages are 'stored' when the app is in background/killed`
so the code is getting triggered.
If you want to show when app is killed in iOS, you have to use `notification`.

**Comment by sarahmarciano** [comment]
Hi,
I have the same problem
when sending a notification without the notification property nothing is received
for ex.
`data: {
            uid: message.user._id,
            type: 'chatSilent',
            message: message.text
          }`
I've also tried adding the `content_available: true` property but it didn't help
BTW this worked for me in the past but I am not sure when it became broken
tnx in advance

**Comment by evollu** [comment]
@sarahmarciano  if it falls apart all of sudden it might be caused by firebase side?
add breakpoints in AppDelegate.m and see if notification ever gets delivered to the phone

**Comment by pacekoding** [comment]
@Ikiugu i spent a whole day to fix this issue, then all works when i changed the dependencies
`//"react-native-fcm": "^9.7.2" 
"react-native-fcm": "6.2.3"`
i hope it helps you

**Comment by n8stowell82** [comment]
so I am going to leave this here for any future people having the same issue. 

TL;DR
make sure you are using the latest Firebase Admin API for your project

I am using the Firebase admin api to send notifications from my server to the devices.  data only messages were working then one day just stopped.  I was researching this last night and saw the suggestion to set breakpoints in the AppDelegate.m file.  Turns out notifications where never making it to my app at all unless I had the `notification` object included.   When digging further into the firebase admin documentation I realized that I was using the legacy api (`sendToDevice()` vs `send()`) when I switched to the newer method I started seeing errors in my console that the Firebase cloud messaging api had not been enabled for this project.  Luckily the error gave me a link to go to so that I could enable it.  Once I followed that link (to another completely different area of google console 🙄 ) I was greeted with a great big "enable api" button.  Now data only messages are working like a charm 😄
